From 9a6546a01b3e87a49f9ce70194a4f6d6656c2cb9 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:44:04 +0530
Subject: [PATCH] system sysnetwork (rebased for 43.6)

Photon does not define distro_redhat. Duplicate the RHEL network file
contexts and sysnet_read_config access under distro_photon so
systemd-networkd and resolved paths are labeled and readable.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/sysnetwork.fc |  9 +++++++++
 policy/modules/system/sysnetwork.if |  6 ++++++
 2 files changed, 15 insertions(+)

diff --git a/policy/modules/system/sysnetwork.fc b/policy/modules/system/sysnetwork.fc
--- a/policy/modules/system/sysnetwork.fc
+++ b/policy/modules/system/sysnetwork.fc
@@ -34,6 +34,15 @@ ifdef(`distro_redhat',`
 /run/systemd/resolve/resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
 /run/systemd/resolve/stub-resolv\.conf  gen_context(system_u:object_r:net_conf_t,s0)
 ')
+
+ifdef(`distro_photon',`
+/etc/sysconfig/network-scripts/.*resolv\.conf -- gen_context(system_u:object_r:net_conf_t,s0)
+/etc/sysconfig/networking(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
+/etc/sysconfig/network-scripts(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
+/run/systemd/network(/.*)?  gen_context(system_u:object_r:net_conf_t,s0)
+/run/systemd/resolve/resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
+/run/systemd/resolve/stub-resolv\.conf  gen_context(system_u:object_r:net_conf_t,s0)
+')
 /run/NetworkManager/resolv\.conf.*   --  gen_context(system_u:object_r:net_conf_t,s0)
 /run/NetworkManager/no-stub-resolv\.conf.*   --  gen_context(system_u:object_r:net_conf_t,s0)
 
diff --git a/policy/modules/system/sysnetwork.if b/policy/modules/system/sysnetwork.if
index f3ea50a..8edff46 100644
--- a/policy/modules/system/sysnetwork.if
+++ b/policy/modules/system/sysnetwork.if
@@ -450,6 +450,12 @@
 		list_dirs_pattern($1, net_conf_t, net_conf_t)
 	')
 
+	ifdef(`distro_photon',`
+		files_search_all_pids($1)
+		init_search_pid_dirs($1)
+		list_dirs_pattern($1, net_conf_t, net_conf_t)
+	')
+
 	ifdef(`distro_redhat',`
 		files_search_all_pids($1)
 		init_search_pid_dirs($1)
